File: AIMToolkit_v4.0.0_rv44/source/AIMLib/AIMLib/entity/NonQuantifiable.cpp
/*
*  2007 – 2013 Copyright Northwestern University
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/


#include "../stdafx.h"
#include "CharacteristicQuantification.h"
#include "NonQuantifiable.h"

#include "../memdebug.h"

using namespace aim_lib;
using iso_21090::CD;

NonQuantifiable::NonQuantifiable(void) : CharacteristicQuantification(C_QUANTIFICATION_NON_QUANTIFIABLE_NAME){
}

NonQuantifiable::NonQuantifiable(const NonQuantifiable& nonQuantifiable) : CharacteristicQuantification(nonQuantifiable){
	*this = nonQuantifiable;
}

NonQuantifiable::~NonQuantifiable(void){
}

NonQuantifiable& NonQuantifiable::operator=(const NonQuantifiable& nonQuantifiable){
	if (this != &nonQuantifiable)
	{
		CharacteristicQuantification::operator=(nonQuantifiable);
		_typeCode = nonQuantifiable._typeCode;
	}
	return *this;
}

NonQuantifiable* NonQuantifiable::Clone() const{
	return new NonQuantifiable(*this);
}

const CharacteristicQuantification::QuantificationType NonQuantifiable::GetQuantificationType() const{
	return QT_Non_Quantifiable;
}

const CD& NonQuantifiable::GetTypeCode() const{
	return _typeCode;
}

void NonQuantifiable::SetTypeCode(const CD& newVal){
	_typeCode = newVal;
}

File: ATB_1.1_src/src/main/java/edu/stanford/isis/atb/domain/template/NonQuantifiable.java
/*
*  Copyright Northwestern University
*  Copyright Stanford University (ATB 1.0 and ATS 1.0)
*
*  Distributed under the OSI-approved BSD 3-Clause License.
*  See http://ncip.github.com/annotation-and-image-markup/LICENSE.txt for details.
*/



package edu.stanford.isis.atb.domain.template;

import org.apache.commons.lang3.StringUtils;
import org.simpleframework.xml.Root;

/**
 * @author Vitaliy Semeshko
 */

@Root(name="NonQuantifiable", strict=false)
public class NonQuantifiable extends CodedTerm {

	@Override
	public String toString() {
		return "Non-Quantifiable: " + StringUtils.trimToEmpty(getCodeMeaning());
	}
	
}
